internal/disk/disk.go

```diff
@@ -47,9 +60,57 @@ type Filesystem struct {
 	FSTabPassNo uint64
 }
 
+// AlignUp will align the given sectors to next aligned sector
+func (pt *PartitionTable) AlignUp(sectors uint64) uint64 {
+
+	grain := pt.BytesToSectors(DefaultGrainBytes)
+
+	return ((sectors + grain) / grain) * grain
+}
+
+// Convert the given bytes to the number of sectors.
+func (pt *PartitionTable) BytesToSectors(size uint64) uint64 {
+	sectorSize := pt.SectorSize
+	if sectorSize == 0 {
+		sectorSize = DefaultSectorSize
+	}
+	return size / sectorSize
+}
+
+// Convert the given number of sectors to bytes.
+func (pt *PartitionTable) SectorsToBytes(size uint64) uint64 {
+	sectorSize := pt.SectorSize
+	if sectorSize == 0 {
+		sectorSize = DefaultSectorSize
+	}
+	return size * sectorSize
+}
+
+// Clone the partition table (deep copy).
+func (pt *PartitionTable) Clone() *PartitionTable {
+	if pt == nil {
+		return nil
+	}
+
+	var partitions []Partition
+	for _, p := range pt.Partitions {
+		p.Filesystem = p.Filesystem.Clone()
+		partitions = append(partitions, p)
+	}
+	return &PartitionTable{
+		Size:       pt.Size,
+		UUID:       pt.UUID,
+		Type:       pt.Type,
+		Partitions: partitions,
+
+		SectorSize:   pt.SectorSize,
+		ExtraPadding: pt.ExtraPadding,
+	}
+}
+
 // Converts PartitionTable to osbuild.QEMUAssemblerOptions that encode
 // the same partition table.
-func (pt PartitionTable) QEMUAssemblerOptions() osbuild.QEMUAssemblerOptions {
+func (pt *PartitionTable) QEMUAssemblerOptions() osbuild.QEMUAssemblerOptions {
```

Comment:
I like to use a non-pointer in these cases because it guarantees immutability.

Reply:
Ah fair point, but since this struct has reference members, it is not _really_ immutability, and there was actually a bug around this in PartitionTable. I also personally think it is surprising if some methods are by-value and some by-ref, but I can adjust if this is what the rest of the code does.

Reply:
+1 for keeping all methods of the same type (by-value or by-type).